### PR TITLE
CA-250748: MTU on pif does not always sync to xapi db

### PR DESCRIPTION
**Pull request by taoyongd**
Signed-off-by: Taoyong Ding <taoyong.ding@citrix.com>

### CODE REVIEW

File: ocaml/xapi/nm.ml [hunk 1]
```diff
@@ -462,18 +462,16 @@ let bring_pif_up ~__context ?(management_interface=false) (pif: API.ref_PIF) =
...
-      end
-    )
+      end;
+
+      (* sync MTU *)
+      try
+        let mtu = Int64.of_int (Net.Interface.get_mtu dbg ~name:bridge) in
+        Db.PIF.set_MTU ~__context ~self:pif ~value:mtu
+      with _ ->
+        debug "could not update MTU field on PIF %s" rc.API.pIF_uuid
```
Comment (lindig) [inline ocaml/xapi/nm.ml]: I think this should have a priority higher than debug - at least info and I would argue warning is appropriate.

Comment (robhoes) [inline ocaml/xapi/nm.ml]: This line wasn't changed, just re-indented. But yes, let's fix it and make it a `warn` now that we are here.

File: ocaml/xapi/nm.ml [hunk 2]
```diff
@@ -462,18 +462,16 @@ let bring_pif_up ~__context ?(management_interface=false) (pif: API.ref_PIF) =
...
-
         Xapi_mgmt_iface.on_dom0_networking_change ~__context
-      end
-    )
+      end;
+
+      (* sync MTU *)
+      try
+        let mtu = Int64.of_int (Net.Interface.get_mtu dbg ~name:bridge) in
+        Db.PIF.set_MTU ~__context ~self:pif ~value:mtu
```
Comment (robhoes) [inline ocaml/xapi/nm.ml]: We can optimise it a little and only do the DB set if the current MTU is different from the one in the DB (`rc.API.pIF_MTU`). This is more important now that this is executed unconditionally.

Comment (taoyongd) [inline ocaml/xapi/nm.ml]: ok, better option

Comment (taoyongd) [inline ocaml/xapi/nm.ml]: updated

**Pull request merged**